platform/security/security-oidc-realm/src/main/java/org/codice/ddf/security/oidc/realm/OidcRealm.java

```diff
@@ -34,53 +39,118 @@
   private static final Logger LOGGER = LoggerFactory.getLogger(OidcRealm.class);
 
   private List<String> usernameAttributeList;
+  private OidcHandlerConfiguration oidcHandlerConfiguration;
 
   /** Determine if the supplied token is supported by this realm. */
   @Override
   public boolean supports(AuthenticationToken token) {
-    boolean supported =
-        token != null && token.getCredentials() != null && token instanceof OidcAuthenticationToken;
-
-    if (supported) {
-      LOGGER.debug("Token {} is supported by {}.", token.getClass(), OidcRealm.class.getName());
-    } else if (token != null) {
-      LOGGER.debug("Token {} is not supported by {}.", token.getClass(), OidcRealm.class.getName());
-    } else {
-      LOGGER.debug("The supplied authentication token is null. Sending back not supported.");
+    if (token instanceof SessionToken) {
+      OidcCredentials credentials = getOidcCredentialsFromSessionToken((SessionToken) token);
+
+      if (credentials == null || credentials.getIdToken() == null) {
+        LOGGER.debug(
+            "The supplied session token didn't have any oidc credentials. Sending back not supported.");
+        return false;
+      }
+      return true;
+    }
+
+    if (!(token instanceof OidcAuthenticationToken)) {
```

Comment:
📘 The only otherauthentication token that the oidc handler can support is an `OidcAuthenticationToken`.